libs/ledger/src/chaincode/smart_contract_wrapper.cpp

```diff
@@ -0,0 +1,32 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ledger/chaincode/smart_contract_wrapper.hpp"
+#include "moment/clocks.hpp"
+
+namespace fetch {
+namespace ledger {
+
+SmartContractWrapper::SmartContractWrapper(ConstByteArray source)
+  : source{std::move(source)}
+  , creation_timestamp{
```

Comment:
We can't have a timestamp in something that will change state. Can we make this a block timestamp please

Reply:
done